packages/bot-skeleton/src/services/tradeEngine/utils/interpreter.js

```diff
@@ -161,16 +173,30 @@ export default class Interpreter {
     }
 
     createAsync(interpreter, func) {
-        return interpreter.createAsyncFunction((...args) => {
+        const asyncFunc = (...args) => {
             const callback = args.pop();
 
-            func(...args.map(arg => interpreter.pseudoToNative(arg)))
+            // Workaround for unknown number of args
+            const reversed_args = args.slice().reverse();
+            const first_defined_arg_idx = reversed_args.findIndex(arg => arg !== undefined);
+
+            // Remove extra undefined args from end of the args
+            const function_args = first_defined_arg_idx < 0 ? [] : reversed_args.slice(first_defined_arg_idx).reverse();
+            // End of workaround
+
+            func(...function_args.map(arg => interpreter.pseudoToNative(arg)))
                 .then(rv => {
                     callback(interpreter.nativeToPseudo(rv));
                     this.loop();
                 })
                 .catch(e => this.$scope.observer.emit('Error', e));
-        });
+        };
+
+        // TODO: This is a workaround, create issue on original repo, once fixed
+        // remove this. We don't know how many args are going to be passed, so we
+        // assume a max of 100.
+        Object.defineProperty(asyncFunc, 'length', { value: 100 });
```

Comment:
Better to define a variable for this, makes it easier to identify:

```suggestion
        Object.defineProperty(asyncFunc, 'length', { value: maxAcceptableFuncArgs });
```

Reply:
That `+ 1` is for the `callback` pushed to the end of the args list.

Reply:
Thank you, I'll add your suggestion. 👍